packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -398,6 +399,19 @@ export class EcsRunTask extends sfn.TaskStateBase implements ec2.IConnectable {
     return policyStatements;
   }
 
+  private getTaskDefinitionArn(): string {
+    const tastDefinitionArn = this.props.taskDefinition.taskDefinitionArn;
+    let needsRevisionWildcard = false;
+    // Check if there is a taskdefinition revision
+    // (arn will end with : followed by digits) included in the arn already
+    if (!cdk.Token.isUnresolved(tastDefinitionArn)) {
```

Comment:
This `EcsRunTask` class does not accept imported task definition (see [code](https://github.com/aws/aws-cdk/pull/31801/files#diff-9d5f521b191b7a78f695205cd6cdbf39fce4ffcae0de03b4c4a92d3c59c06f3cR29)). So we can skip this `if` block.

Reply:
Good point. Will remove this `if` block.